#VRML_SIM R2020a utf8
WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation 0.9992054075307898 0.03242925583463059 0.023171036386842637 5.041636866658491
  position -0.7962251117874557 6.907797450221645 2.3802587455690407
  followType "Pan and Tilt Shot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 3.66 3.66
  tileSize 3.66 3.66
  appearance PBRAppearance {
    baseColor 0.9 0.9 0.9
    baseColorMap ImageTexture {
      url [
        "test_arena_B-textures/test_arena_B-search_and_beaconing.png"
      ]
      repeatS FALSE
      repeatT FALSE
    }
    roughness 1
    metalness 0
  }
}
DEF ARENA_WALLS Group {
  children [
    Solid {
      translation 1.79 0.15 0.04
      children [
        DEF ARENA_WALL_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.701961 0.701961 0.701961
            roughness 1
          }
          geometry DEF ARENA_WALL_GEOM Box {
            size 0.08 0.3 3.58
          }
        }
      ]
      name "wall1"
      boundingObject USE ARENA_WALL_GEOM
    }
    Solid {
      translation -0.04 0.15 1.79
      rotation 0 1 0 1.5707996938995747
      children [
        USE ARENA_WALL_SHAPE
      ]
      name "wall2"
      boundingObject USE ARENA_WALL_GEOM
    }
    Solid {
      translation -1.79 0.15 -0.04
      children [
        USE ARENA_WALL_SHAPE
      ]
      name "wall3"
      boundingObject USE ARENA_WALL_GEOM
    }
    Solid {
      translation 0.04 0.15 -1.79
      rotation 0 1 0 -1.5707996938995747
      children [
        USE ARENA_WALL_SHAPE
      ]
      name "wall4"
      boundingObject USE ARENA_WALL_GEOM
    }
  ]
}
DEF MAZE_WALLS Group {
  children [
    MazeWall {
      translation -1.357 0.15 0.779
      rotation 0 -1 0 -5.307179586466759e-06
      name "maze wall custom(30)"
      length 0.75
    }
    MazeWall {
      translation -0.401 0.15 -0.574
      rotation 0 -1 0 -5.307179586466759e-06
      name "maze wall custom(1)"
      length 0.95
    }
    MazeWall {
      translation 0.467 0.15 -0.276
      rotation 0 -1 0 0.6544936938995747
      name "maze wall custom(2)"
      length 0.95
    }
    MazeWall {
      translation 0.848 0.15 0.302
      rotation 0 -1 0 1.5707896938995747
      name "maze wall custom(3)"
      length 0.55
    }
    DEF ZONE_A_WALLS Group {
    }
    DEF ZONE_B_WALLS Group {
    }
    DEF ZONE_C_WALLS Group {
    }
  ]
}
DEF BEACONS Group {
  children [
    DEF RED_CYL Solid {
      translation -0.0759672 0.2 -0.285755
      children [
        DEF BCN_RED Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.4
            radius 0.15
          }
        }
      ]
      boundingObject USE BCN_RED
      physics DEF BCN_PHYS Physics {
        density -1
        mass 0.5
      }
      recognitionColors [
        1 0 0
      ]
    }
    DEF FCA_CYL Solid {
      translation -0.356726 0.1 -1.51029
      children [
        DEF BCN_FCA Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.2
            radius 0.2
          }
        }
      ]
      name "solid(2)"
      boundingObject USE BCN_FCA
      physics USE BCN_PHYS
      recognitionColors [
        1 0 1
      ]
    }
    DEF TEAL_BOX Solid {
      translation 1.45023 0.2 1.47189
      children [
        DEF BCN_TEAL Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.45 0.4 0.45
          }
        }
      ]
      name "solid(1)"
      boundingObject USE BCN_TEAL
      physics USE BCN_PHYS
      recognitionColors [
        0 0.5 0.5
      ]
    }
    DEF GRN_RECT Solid {
      translation 0.701167 0.15 -0.375253
      rotation 0 1 0 -0.6544996938995746
      children [
        DEF BCN_GRN Shape {
          appearance PBRAppearance {
            baseColor 0 0.5 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.6 0.3 0.3
          }
        }
      ]
      name "solid(3)"
      boundingObject USE BCN_GRN
      physics USE BCN_PHYS
      recognitionColors [
        0 0.5 0
      ]
    }
    DEF BLU_RECT Solid {
      translation -1.42411 0.11 0.336374
      rotation 0 1 0 1.0471996938995747
      children [
        DEF BCN_BLU Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.6 0.22 0.25
          }
        }
      ]
      name "solid(4)"
      boundingObject USE BCN_BLU
      physics USE BCN_PHYS
      recognitionColors [
        0 0 0.5
      ]
    }
  ]
}
StartZone {
  translation -1.47 0 1.488
  rotation 0 1 0 3.141592653589793
  colour 0 0.5 0
  name "start zone A"
  recColSlot [
    0 0.5 0
  ]
}
StartZone {
  translation -0.865 0 -1.47
  rotation 0 1 0 -1.5707996938995747
  colour 0 0.5 0.5
  name "start zone B"
  recColSlot [
    0 0.5 0.5
  ]
}
StartZone {
  translation 1.471 0 -1.489
  colour 0 0 0.5
  name "start zone C"
  recColSlot [
    0 0 0.5
  ]
}
Robot {
  translation -1.47 0.0397205 1.471
  rotation 7.223109997514686e-06 0.9999999996559219 -2.5218699991322798e-05 3.14
  children [
    GPS {
    }
    Camera {
      translation 0 0.03 0.085
      rotation 0 1 0 3.14159
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.04 0.01 0.01
          }
        }
      ]
      name "camera0"
      width 100
      height 100
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.678431 0.498039 0.658824
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF TL_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 500.09571190081215
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "tlwheel"
        }
      ]
      endPoint Solid {
        translation 0.059999999603073784 -2.8544171330280033e-06 0.05000269263808757
        rotation -0.7059329956090296 0.7059355887073491 0.05756344592595712 3.026593090711371
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.203922 0.396078 0.643137
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF TR_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 452.14101668453543
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "trwheel"
        }
      ]
      endPoint Solid {
        translation -0.0600000003974106 3.595854379942165e-06 0.04999842920827169
        rotation -0.4933293044325196 0.49333111663075263 -0.7164151078473856 4.384905315914681
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF BL_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 500.0957365388898
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "blwheel"
        }
      ]
      endPoint Solid {
        translation 0.060000000397222086 -2.8543884036839627e-06 -0.049997307331450705
        rotation -0.7059329460986935 0.7059355392145876 0.05756466004614832 3.026590670473263
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF BR_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        position 452.14104124762804
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "brwheel"
        }
      ]
      endPoint Solid {
        translation -0.05999999960326252 3.595837682641646e-06 -0.050001570829962615
        rotation -0.49333021970639246 0.4933320318882275 -0.71641384732393 4.384903649909941
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.0697959 0.01 0.0574741
      rotation 0 1 0 3.12165
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.988235 0.913725 0.309804
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      rotationStep 0.262
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 2000 0
        0.3 3000 0
        0.4 4000 0
        0.5 5000 0
        0.56 5600 0
      ]
      type "laser"
      aperture 0.01
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.0698 0.01 0.0575
      rotation 0 1 0 0.176175
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.988235 0.913725 0.309804
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      rotationStep 0.262
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 2000 0
        0.3 3000 0
        0.4 4000 0
        0.5 5000 0
        0.56 5600 0
      ]
      type "laser"
      aperture 0.01
    }
    DEF DS_FRONT DistanceSensor {
      translation -0.005 0 0.1
      rotation 0 1 0 -1.8707853071795864
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.988235 0.913725 0.309804
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_front"
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 2000 0
      ]
      type "laser"
      aperture 0.01
    }
  ]
  name "team3_robot_Cyrano"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "robot_control"
}
